tests/Functional/Ticket/GH181Test.php

```diff
@@ -0,0 +1,67 @@
+<?php
+declare(strict_types=1);
+
+namespace KafkaTest\Functional\Ticket;
+
+use Kafka\Exception\InvalidRecordInSet;
+use Kafka\Producer;
+use Kafka\ProducerConfig;
+use PHPUnit\Framework\TestCase;
+use function getenv;
+
+final class GH181Test extends TestCase
+{
+    public function setUp(): void
+    {
+        $version = getenv('KAFKA_VERSION');
+        $brokers = getenv('KAFKA_BROKERS');
+
+        if (! $version || ! $brokers) {
+            self::markTestSkipped(
+                'Environment variables "KAFKA_VERSION" and "KAFKA_BROKERS" must be provided'
+            );
+        }
+
+        $config = ProducerConfig::getInstance();
+        $config->setMetadataBrokerList($brokers);
+        $config->setBrokerVersion($version);
+
+        parent::setUp();
+    }
+
+    /**
+     * @dataProvider dataProviderTestInvalidRecordSet
+     * @group GH181
```

Comment:
Sorry, I just saw this now... let's keep only the ticket number here to make things simpler.

Reply:
Don't have to apologise, I'm grateful for reviews ;)

So when you request new changes, doesn't make me mad :P

Reply:
I'll give plenty of reviews here HAHAHA

Reply:
Epic 👍 😂